lib/pf/cmd/pf/service.pm

```diff
@@ -0,0 +1,359 @@
+        if(all { $_->status eq '0'  } @managers) {
+            $technique = getIptablesTechnique();
+            $technique->iptables_save( $install_dir . '/var/iptables.bak' );
+        }
+        $technique ||= getIptablesTechnique();
+        $technique->iptables_generate();
+    }
+
+    my ($noCheckupManagers,$checkupManagers) = part { $_->shouldCheckup} @managers;
+
+    if($noCheckupManagers && @$noCheckupManagers) {
+        foreach my $manager (@$noCheckupManagers) {
+            _doStart($manager);
+        }
+    }
+    if($checkupManagers && @$checkupManagers) {
+        checkup( map {$_->name} @$checkupManagers);
+        foreach my $manager (@$checkupManagers) {
+            _doStart($manager);
+        }
+    }
+    return 0;
+}
+
+sub checkup {
+    require pf::services;
+    require pf::pfcmd::checkup;
+    no warnings "once"; #avoids only used once warnings generated by the access of pf::pfcmd::checkup namespace
+    my @services;
+    if(@_) {
+        @services = @_;
+    } else {
+        @services = @pf::services::ALL_SERVICES;
+    }
+
+    my @problems = pf::pfcmd::checkup::sanity_check(pf::services::service_list(@services));
+    foreach my $entry (@problems) {
+        chomp $entry->{$pf::pfcmd::checkup::MESSAGE};
+        print $entry->{$pf::pfcmd::checkup::SEVERITY}  . " - " . $entry->{$pf::pfcmd::checkup::MESSAGE} . "\n";
+    }
+
+    # if there is a fatal problem, exit with status 255
+    foreach my $entry (@problems) {
+        if ($entry->{$pf::pfcmd::checkup::SEVERITY} eq $pf::pfcmd::checkup::FATAL) {
+            exit(255);
+        }
+    }
+
+    if (@problems) {
+        return $TRUE;
+    } else {
+        return $FALSE;
+    }
+}
+
+sub _doStart {
+    my ($manager) = @_;
+    my $command;
+    my $color = '';
+    if($manager->status ne '0') {
+        $color =  $WARNING_COLOR;
+        $command = 'already started';
+    } else {
+        if($manager->start) {
+            $command = 'start';
+            $color =  $SUCCESS_COLOR;
+        } else {
+            $command = 'not started';
+            $color =  $ERROR_COLOR;
+        }
+    }
+    print $manager->name,"|${color}${command}${RESET_COLOR}\n";
+}
+
+sub getManagers {
+    my ($services,$flags) = @_;
+    $flags = 0 unless defined $flags;
+    my %seen;
+    my $includeDependsOn = $flags & INCLUDE_DEPENDS_ON;
+    my $justManaged      = $flags & JUST_MANAGED;
+    my @serviceManagers =
+        grep { (!exists $seen{$_->name}) && ($seen{$_->name} = 1) && ( !$justManaged || $_->isManaged ) }
+        map {
+            my $m = $_;
+            my @managers =
+                grep { defined $_ }
+                map { pf::services::get_service_manager($_) }
+                @{$m->dependsOnServices}
+                if $includeDependsOn;
+            push @managers,$m;
+            @managers
+        }
+        grep { defined $_ }
+        map { pf::services::get_service_manager($_) } @$services;
+    return @serviceManagers;
```

Comment:
You are trying just a little too hard to impress us poor mortals.
A twelve line grep | map | grep | map pipeline?
Please, this is not lisp ;-)
Refactor this to make it readable to the rest of us.